Dockerfile

```diff
@@ -36,12 +36,11 @@ RUN . $VENV_DIR/bin/activate && pip install setuptools==75.1.0
 # [VULN-XXX] update libcap2 to 1:2.66-4+deb12u1
 # [VULN-613] update systemd to 252.38-1~deb12u1.
 RUN apt-get update \
-    && apt-get install -y gnupg gnupg2 gnupg1 curl apt-transport-https \
-    && curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - \
-    && curl https://packages.microsoft.com/config/debian/10/prod.list \
-    > /etc/apt/sources.list.d/mssql-release.list \
+    && apt-get install -y gnupg gnupg2 gnupg1 curl apt-transport-https ca-certificates \
+    && curl -fsSL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor -o /usr/share/keyrings/microsoft-archive-keyring.gpg \
+    && echo "deb [arch=amd64,arm64,armhf signed-by=/usr/share/keyrings/microsoft-archive-keyring.gpg] https://packages.microsoft.com/debian/12/prod bookworm main" > /etc/apt/sources.list.d/mssql-release.list \
```

Comment:
could you leave a comment in the file explaining what these commands do and why we need them?